File: Solutions/0191/0191.java
/**
 * @Author: washing
 * @DateTime: 2021/01/23 23:07
 * @File: 0191.java
 * @Desc: 
 **/

public class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {
        int count = 0;
        while(n!=0) {
            n&=(n-1);
            count++;
        }
        return count;
    }
}

File: Solutions/0191/0191.c
// Author: washing
// DateTime: 2021/3/22 10:24
// File: 0191.c
// Desc: 

int hammingWeight(uint32_t n) {
    int a=0;
    while (n > 0){
        a += n%2;
        n /= 2;
    }
    return a;
}